### PR TITLE
events block Russian description Close #755

### LINKED ISSUE #755: Russian documentation for events block

### PR DESCRIPTION
**Pull request by Murdalay**

### CODE REVIEW

File: common.blocks/events/events.ru.md [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+    
+    events.Emitter.emit(myevent);               // 'bar from e.data'
+
+provide({ 'subscribe' : subscribe });
+});
+```
+
+Как видно из примера, при использовании готового объекта события, значение аргумента `[data]` не присваивается полю `e.data` объекта.
+
+Результаты выполнения функции-обработчика сохраняются в поле `e.res` соответствующего объекта события.
```
Comment (qfox) [inline common.blocks/events/events.ru.md]: `res` или `result`?

File: common.blocks/events/__channels/events__channels.ru.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
+# events__channels
+
+Элемент `channels` блока `events` предназначен для работы с именными каналами событий. Именные каналы позволяют организовать работу с событиями, используя шаблон проектирования «наблюдатель» (также известный как Publisher-subscriber).
```
Comment (tadatuta) [inline common.blocks/events/__channels/events__channels.ru.md]: именованные

File: common.blocks/events/__channels/events__channels.ru.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+Элемент реализует функцию, позволяющую:
+
+* получить ссылку на именной канал по `id`.
```
Comment (tadatuta) [inline common.blocks/events/__channels/events__channels.ru.md]: . -> ;

File: common.blocks/events/__channels/events__channels.ru.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+Элемент `events` реализован в технологии `vanila.js` и подходит для использования как на клиенте, так и на сервере.
+
+Функция принимает аргументы:
+
+* `{String}` `[id]` – Идентификатор канала. Если не задан будет использоваться канал по умолчанию (`'default'`).
+* `{Boolean}` `[drop]` – Логический флаг, указывающий (в значении `true`) на необходимость удалить канал. По умолчанию `false`.
+
+Функция возвращает:
+* ссылку на объект «класса» `Emitter` – именной канал.
+* `undefined` если функция была вызвана с параметром `drop` в значении `true`.
```
Comment (tadatuta) [inline common.blocks/events/__channels/events__channels.ru.md]: ,

File: common.blocks/events/__channels/events__channels.ru.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+```js
+modules.define('channels-test', ['events__channels'], function(provide, channels) {
```
Comment (tadatuta) [inline common.blocks/events/__channels/events__channels.ru.md]: жизненный пример

File: common.blocks/events/events.ru.md [hunk 6]
```diff
@@ -0,0 +1,309 @@
+# events
+
+Блок предоставляет механизмы для работы с событиями JavaScript, реализованные в виде псевдоклассов:
+
+* `Event` – для создания объекта события, изменения и проверки его состояний.
```
Comment (tadatuta) [inline common.blocks/events/events.ru.md]: ;

File: common.blocks/events/events.ru.md [hunk 7]
```diff
@@ -0,0 +1,309 @@
...
+* `Emitter` – для генерации событий и подписки на них.
+
+Блок `events` реализован в технологии `vanila.js` и подходит для использования как на клиенте, так и на сервере.
+
+Блок служит основой механизма событий в `bem-core`. Базовый блок `BEM` наследуется от «класса» `Emitter`. Поэтому нет необходимости использовать `events`, если вы подключаете блок `i-bem`.
```
Comment (tadatuta) [inline common.blocks/events/events.ru.md]: не только

File: common.blocks/events/events.ru.md [hunk 8]
```diff
@@ -0,0 +1,309 @@
...
+* `Event` – для создания объекта события, изменения и проверки его состояний.
+* `Emitter` – для генерации событий и подписки на них.
+
+Блок `events` реализован в технологии `vanila.js` и подходит для использования как на клиенте, так и на сервере.
+
+Блок служит основой механизма событий в `bem-core`. Базовый блок `BEM` наследуется от «класса» `Emitter`. Поэтому нет необходимости использовать `events`, если вы подключаете блок `i-bem`. 
+
+## `Event`
+
+С помощью псевдокласса можно инстанцировать объект события, указав при необходимости тип и триггер события:
```
Comment (tadatuta) [inline common.blocks/events/events.ru.md]: избавиться от тавтологии (его тип и триггер)

File: common.blocks/events/events.ru.md [hunk 9]
```diff
@@ -0,0 +1,309 @@
...
+
+```js
```
Comment (tadatuta) [inline common.blocks/events/events.ru.md]: стоит выпилить пример

File: common.blocks/events/events.ru.md [hunk 10]
```diff
@@ -0,0 +1,309 @@
...
+Функция-конструктор, выполняемая в ходе инстанцирования объекта события, принимает аргументы:
+
+* `type` `{String}` – тип события.
+* `target` `{Object}` – объект (триггер) на котором событие возникло.
+
+Значения аргументов присваиваются одноименным полям объекта событий. Кроме указанных, новый объект событий содержит следующий набор полей: 
+
+* `{*}` `result` – по умолчанию `undefined`. Поле для сохранения результата выполнений функции-обработчика события.
+* `{*}` `data` – по умолчанию `undefined`. Поле для хранения произвольных данных, передаваемых обработчику события.
+* `{Boolean}` `_isDefaultPrevented` – по умолчанию `false`. Предотвращать стандартное действие для HTML-элемента (со значением `true`).
```
Comment (tadatuta) [inline common.blocks/events/events.ru.md]: не описываем приватные поля

File: common.blocks/events/events.ru.md [hunk 11]
```diff
@@ -0,0 +1,309 @@
...
+* `{Boolean}` `_isPropagationStopped` – по умолчанию `false`. Остановить трансляцию события (со значением `true`).
+
+### Методы объекта события
+
+#### `preventDefault`
+
+Служит для предотвращения выполнения стандартного действия, заданного для HTML-элемента. Фактически, метод устанавливает полю `_isDefaultPrevented` значение `true`. Метод не принимает аргументов.
```
Comment (tadatuta) [inline common.blocks/events/events.ru.md]: для HTML-элемента -> для события

выпилить фразу «Фактически...»

File: common.blocks/events/events.ru.md [hunk 12]
```diff
@@ -0,0 +1,309 @@
...
+
+```js
+modules.define('event-test', ['events'], function(provide, events) {
```
Comment (tadatuta) [inline common.blocks/events/events.ru.md]: нужен жизненный пример

File: common.blocks/events/events.ru.md [hunk 13]
```diff
@@ -0,0 +1,309 @@
...
+    console.log(myevent.isDefaultPrevented()); // true
+
+provide(myevent);
+});
+```
+
+
+#### `stopPropagation` 
+
+Служит для остановки трансляции события. Фактически, метод устанавливает полю `_isPropagationStopped` значение `true`. Метод не принимает аргументов.
```
Comment (tadatuta) [inline common.blocks/events/events.ru.md]: не писать про _isPropagationStopped

File: common.blocks/events/events.ru.md [hunk 14]
```diff
@@ -0,0 +1,309 @@
...
+    console.log(myevent.isPropagationStopped()); // true
+
+provide(myevent);
+});
+```
+
+
+## `Emitter`
+
+С помощью псевдокласса можно генерировать события и осуществлять подписку на них. Это производится с помощью методов «класса».
```
Comment (tadatuta) [inline common.blocks/events/events.ru.md]: убрать кавычки, подумать, про переименование псевдоклассов в классы

File: common.blocks/events/events.ru.md [hunk 15]
```diff
@@ -0,0 +1,309 @@
...
+
+### Методы «класса»
+
+Набор и сигнатуры статических методов идентичны набору и сигнатурам методов объекта, инстанцируемого классом.
+
+#### `on`
+
+Служит для подписки на событие определенного типа. Метод принимает следующие аргументы:
+
+ * `{String}` `e` – тип события, на которое производится подписка.
```
Comment (tadatuta) [inline common.blocks/events/events.ru.md]: e -> type

File: common.blocks/events/events.ru.md [hunk 16]
```diff
@@ -0,0 +1,309 @@
...
+provide({ 'runOnce' : runOnce });
+});
+```
+
+
+#### `un`
+
+Служит для удаления установленной ранее функции-обработчика события. Метод принимает следующие аргументы:
+
+ * `{String}` `e` – тип события у которого удаляется обработчик.
```
Comment (tadatuta) [inline common.blocks/events/events.ru.md]: ;

File: common.blocks/events/events.ru.md [hunk 17]
```diff
@@ -0,0 +1,309 @@
...
+
+Служит для генерации события. Другими словами, метод вызывает все функции-обработчики, заданные для события, переданного аргументом. Метод принимает следующие аргументы:
+
+ * `{String|events:Event}` `e` – генерируемое событие в виде строки или ссылки на готовый объект события.
+ * `{Object}` `[data]` – дополнительные данные, доступные как второй аргумент функции-обработчика.
+
+Метод возвращает ссылку на объект `this`.
+
+```js
+modules.define('event-test', ['events'], function(provide, events) {
```
Comment (tadatuta) [inline common.blocks/events/events.ru.md]: придумать пример для этого поведения

**Comment by Murdalay** [comment]
@dfilatov I really need some help with examples. 

There are few questions in order of importance:
- `events.Event`
  - should I use `Event` constructor at all, and if so, for what purpose most likely
  - can you help me to figure out a real example of `preventDefault` usage
- `events.Emitter`
  - why we need static methods along with own ones
  - is there any broad examples for own and static methods usage
  - why, if i subscribe on certain event type with static method, it will ignore that event type fired from own one, and vice versa 
  - can you tell something about relations of `e.data` field and second argument of `emit` method – when it redefines `e.data` field and why
  - when should I pass a ready event object instead of a string value

**Comment by dfilatov** [comment]
You should use  `events.Event` in cases you need to interact with event object after handlers are applied. Most of cases are checking for `event.isDefaultPrevented()` or `event.isPropagationStopped()`.
Example: you have a block `link` which emits `click` event and navigates to a given url. Handlers are able to prevent this nagivation by calling `event.isDefaultPrevented()`.

**Comment by dfilatov** [comment]
> can you tell something about relations of e.data field and second argument of emit method – when it redefines e.data field and why

The second argument of `emit` never redefines `e.data`. `e.data` is a data which is passed when you're subscribing to emitter.

**Comment by dfilatov** [comment]
> why, if i subscribe on certain event type with static method, it will ignore that event type fired from own one, and vice versa 

Sorry, I haven't understood. Could you provide example?

**Comment by Murdalay** [comment]
@dfilatov for example:

``` js
modules.require(['events'], function(events) {
    var myemmiter = new events.Emitter(),
        _shout = function(){ console.log('foo') },
        _yeld = function(){ console.log('bar') };

    events.Emitter.on('myevent', _shout); 
    myemmiter.on('myevent', _yeld);

    events.Emitter.emit('myevent'); // 'foo'
    myemmiter.emit('myevent'); // 'bar'

});
```

**Comment by dfilatov** [comment]
What would you expect in such case?

**Comment by Murdalay** [comment]
I mean, in which cases should I use static methods and in which own ones?

**Comment by dfilatov** [comment]
You should use instance method if you want emit event on instance and use "class" if you want emit event on "class".

**Comment by qfox** [comment]
:+1: Water for drinking, food for eating.

**Comment by Murdalay** [comment]
@zxqfox +1 )

**Comment by Murdalay** [comment]
I rewrite the example for `events__channels` and fix grammar. Still need to rewrite last example.

File: common.blocks/events/events.ru.md [hunk 18]
```diff
@@ -0,0 +1,389 @@
...
+
+<a name="fields-result"></a>
+##### Свойство `result`
+
+Тип: `{*}`.
+
+Содержит данные, возвращаемые последней функцией-обработчиком события.
+
+```js
+modules.require(['events'], function(events) {
```
Comment (narqo) [inline common.blocks/events/events.ru.md:98]:
``` js
modules.require(['events'], function(events) {

var myEmitter = new events.Emitter();
myEmitter.on('myevent', function() { return 'hi-hi-hi'; });

var myEvent = new events.Event('myevent');
myEmitter.emit(myEvent)

console.log(myEvent.result);    // 'hi-hi-hi'

});
```

File: common.blocks/events/events.ru.md [hunk 19]
```diff
@@ -0,0 +1,389 @@
...
+
+<a name="fields-data"></a>
+##### Свойство `data`
+
+Тип: `{*}`.
+
+Содержит данные, передаваемые функции-обработчику события в качестве аргумента.
+
+```js
+modules.require(['events'], function(events) {
```
Comment (narqo) [inline common.blocks/events/events.ru.md:119]:
``` js
modules.require(['events'], function(events) {

var myEmitter = new events.Emitter();
myEmitter.on('myevent', 'my-data', function(e) { console.log(e.data); });

myEmitter.emit('myevent'); // my-data

});
```

File: common.blocks/events/__channels/events__channels.ru.md [hunk 20]
```diff
@@ -0,0 +1,43 @@
...
+
+* [`{String}` `id`] – Идентификатор канала. Если не задан будет использоваться канал по умолчанию (`'default'`).
+* [`{Boolean}` `drop`] – Логический флаг, указывающий (в значении `true`) на необходимость удалить канал. По умолчанию `false`.
+
+Возвращаемое значение:
+
+* `{Object}`. Объект «класса» `Emitter` – именной канал.
+* `{undefined}`. В случае если функция была вызвана с параметром `drop` в значении `true`.
+
+```js
```
Comment (narqo) [inline common.blocks/events/__channels/events__channels.ru.md:21]:
``` js
modules.require(['events__channels'], function(channels) {

var myChannel = channels('my-channel');
myChannel.on('test', function(e, data) { console.log(data.foo) });

myChannel.emit('test', { foo : 'bar' }); // 'bar'

});
```

**Comment by Murdalay** [comment]
Merge me, please.

**Pull request merged**